src/components/utils/api.js

```diff
@@ -19,7 +19,85 @@ export const getCommentsByArticleId = (id) => {
 };
 
 export const patchVote = (path, vote) => {
-  return articlesAPI
-    .patch(path, { inc_votes: vote })
-    .catch((err) => console.log(err));
+  return articlesAPI.patch(path, { inc_votes: vote });
+};
+
+export const errorTimeout = ({
+  setDoubleClickUp,
+  setDoubleClickDn,
+  setError,
+}) => {
+  setTimeout(() => {
+    setDoubleClickUp(false);
+    setDoubleClickDn(false);
+    setError(null);
+  }, 2000);
+};
+
+export const changeVote = (
+  e,
+  path,
+  setVotes,
+  setPreviousClick,
+  previousClick,
+  setDoubleClickUp,
+  setDoubleClickDn,
+  setError
+) => {
+  e.preventDefault();
+  const {
+    target: { value },
+  } = e;
+  setVotes((prevVotes) => prevVotes + +value);
+  setPreviousClick((prePre) => +prePre + +value);
+  if (value > 0 && previousClick === 0) {
+    (() => {
+      setDoubleClickUp(true);
+      setDoubleClickDn(false);
+    })();
+  }
+  if (value > 0 && previousClick === -1) {
+    (() => {
+      setDoubleClickUp(false);
+      setDoubleClickDn(false);
+    })();
+  }
+  if (value < 0 && previousClick === 0) {
+    (() => {
+      setDoubleClickUp(false);
+      setDoubleClickDn(true);
+    })();
+  }
+  if (value < 0 && previousClick === 1) {
+    (() => {
+      setDoubleClickUp(false);
+      setDoubleClickDn(false);
+    })();
+  }
+  patchVote(path, value).catch(({ message: err }) => setError(err));
+};
+
+export const handleSubmit = (
+  e,
+  commentBody,
+  userName,
+  path,
+  setArticleComments
+) => {
+  e.preventDefault();
+  if (commentBody.length > 3) {
+    const date = new Date();
+    let userComment = {
+      author: userName,
+      created_at: date.toString(),
+      votes: 0,
+      body: commentBody,
+    };
+    setArticleComments((currentComments) => [userComment, ...currentComments]);
+
```

Comment:
Nice updating of state alongside the API call, this function would be fine to include in the PostComment component rather than here, and would maybe be a bit easier to follow there
You could also add some kind of message to tell the user why their comment hasn't gone through (i.e. that their comment isn't at least 3 characters long)